azure/storage/blob/baseblobservice.py

```diff
@@ -240,6 +244,9 @@ def make_blob_url(self, container_name, blob_name, protocol=None, sas_token=None
             blob_name,
         )
 
+        if snapshot:
+            url += '?snapshot=' + snapshot
```

Comment:
in the situation that both `sas_token` and `snapshot` are both given, this simple approach will cause duplicate `?` and result a bad formed URL.